core/src/impl/Kokkos_Core.cpp

```diff
@@ -767,7 +770,8 @@ void post_initialize_internal(const Kokkos::InitializationSettings& settings) {
   Kokkos::Tools::InitArguments tools_init_arguments;
   combine(tools_init_arguments, settings);
   initialize_profiling(tools_init_arguments);
-  g_is_initialized = true;
+  g_is_initialized          = true;
+  g_initialization_settings = settings;
```

Comment:
At this point, does `settings` contain information about settings that are implicitly set? For example, if the user doesn't set num threads (for a CPU device), does _settings_ have the number of threads that will actually be used?

Reply:
No.  It contains the settings resolved from the argument passed to `Kokkos::initialize` and the environment variables.

Reply:
Is there a way to fill it with that information? That is something we would like to capture if possible.

Reply:
Can you elaborate on why the version proposed is not sufficient?
If `device_id` is set, you will get that device or Kokkos will call abort if it is not able to.  We will not just decide to pick another one.  Similarly, if `num_threads` is specified, then Kokkos will honor that choice.

Reply:
The point is to get a hint about what kokkos is doing when something is _not_ set. If I don't set `num_threads`, how many threads will Kokkos use? The answer is not obvious. Will it use the number of physical threads, or will it include hyperthreads? Did a job scheduler partition threads in a way that reduces them per process?

Likewise, if I _don't_ set `device_id`, I might want to get an indication of which id Kokkos picked. Did the job scheduler properly allocate different devices for each process on the same node, or did I dork something up with the scheduling flags?

Reply:
OK.  I would explore another direction for this.  I don't like the idea of our returning `InitializationSettings` for that purpose.  Let's take back the discussion to #5843

I will keep that draft open a bit longer to see if there is interest in that facility.